third_party/nvfuser/python/__init__.py

```diff
@@ -2,26 +2,89 @@
 from . import _C
 from ._C import *
 
-class FusionDefinition(_C._FusionDefinition):
+import numbers
+from functools import partial
+
+class PythonOperators:
+    def _not_number_scalar(self, num):
+        return not isinstance(num, numbers.Number) or isinstance(num, bool)
+    def _convert_python_number(self, op, *args, **kwargs):
+        # TODO: handle python containers as well.
+        args = [
+            x if self._not_number_scalar(x) else self._fd.define_constant(x)
+            for x in args
+        ]
+        for k, v in kwargs.items():
+            if not self._not_number_scalar(v):
+                kwargs[k] = self._fd.define_constant(v)
+        return op(*args, **kwargs)
+
+    def bindOperation(self, op_name: str, op):
+        # TODO remove this after we can handle python containers.
+        if op_name not in ["var", "var_mean"]:
```

Comment:
This is ugly!

The reason is that we don't yet support vector on scalar. I'll do that in a separate PR and we should be able to remove this. I'm adding this hack so we are not changing API on var/var_mean.